frontend/src/main/scala/bloop/engine/Interpreter.scala

```diff
@@ -252,10 +251,32 @@ object Interpreter {
     state.mergeStatus(ExitStatus.InvalidCommandLineOption)
   }
 
-  private def waitAndLog(previousState: State, newState: Task[State]): State = {
+  private def waitAndLog(
+      action: Action,
+      cliOptions: CliOptions,
+      previousState: State,
+      newState: Task[State]
+  ): State = {
+    val pool = previousState.pool
+    val ngout = cliOptions.common.ngout
     try {
-      // Duration has to be infinity, we cannot predict how much time compilation takes
-      Await.result(newState.runAsync(previousState.scheduler), Duration.Inf)
+      val handle = newState
+        .executeWithOptions(_.enableAutoCancelableRunLoops)
+        .runAsync(previousState.scheduler)
+
+      // Let's cancel tasks (if supported by the underlying implementation) when clients disconnect
+      pool.addListener {
+        case e: CloseEvent =>
+          if (!handle.isCompleted) {
+            System.out.println(
```

Comment:
So that's writing to stdout of the server? Isn't that what `ngout` is for?

Reply:
Indeed, but in this case the stdout that we see here is already ngout! I'll change it though, for clarity :smile: